File: fe2-android/app/src/main/java/com/github/dedis/popstellar/repository/database/digitalcash/TransactionDao.java
package com.github.dedis.popstellar.repository.database.digitalcash;

import androidx.room.*;

import com.github.dedis.popstellar.model.objects.digitalcash.TransactionObject;

import java.util.List;

import io.reactivex.Completable;
import io.reactivex.Single;

@Dao
public interface TransactionDao {

  @Insert(onConflict = OnConflictStrategy.REPLACE)
  Completable insert(TransactionEntity transactionEntity);

  @Query("SELECT `transaction` FROM transactions WHERE lao_id = :laoId")
  Single<List<TransactionObject>> getTransactionsByLaoId(String laoId);

  @Query("DELETE FROM transactions WHERE lao_id = :laoId")
  Completable deleteByLaoId(String laoId);
}

File: fe2-android/app/src/main/java/com/github/dedis/popstellar/repository/database/digitalcash/TransactionDao.kt
package com.github.dedis.popstellar.repository.database.digitalcash

import androidx.room.Dao
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query
import com.github.dedis.popstellar.model.objects.digitalcash.TransactionObject
import io.reactivex.Completable
import io.reactivex.Single

@Dao
interface TransactionDao {
  @Insert(onConflict = OnConflictStrategy.REPLACE)
  fun insert(transactionEntity: TransactionEntity): Completable

  @Query("SELECT `transaction` FROM transactions WHERE lao_id = :laoId")
  fun getTransactionsByLaoId(laoId: String): Single<List<TransactionObject>?>

  @Query("DELETE FROM transactions WHERE lao_id = :laoId")
  fun deleteByLaoId(laoId: String): Completable
}
